packages/core/src/context/context.ts

```diff
@@ -59,45 +62,79 @@ export const registerAll = (boundDependencies: BoundDependency<any, any>[]) => (
     context,
   );
 
+/**
+ * Resolves eager dependencies within the context
+ * @since v2.0.0
+ */
 export const resolve = async (context: Context): Promise<Context> => {
+  const resolveEagerDependency = ({ token, dependency }: BoundDependency<unknown, ContextEagerReader>): T.Task<Context> => pipe(
+    () => pipe(context, dependency.eval, d => Promise.resolve(d)),
```

Comment:
`pipe(context, dependency.eval, Promise.resolve)`

Reply:
It won't work because `Promise.resolve` is not a unary function

Reply:
Well... it doesn't work for a different reason. `Promise.resolve` is evaluated too early